distributedcombigrid/src/sgpp/distributedcombigrid/sparsegrid/DistributedSparseGridUniform.hpp

```diff
@@ -134,7 +194,16 @@ class DistributedSparseGridUniform {
 
   int commSize_;
 
-  std::vector<SubspaceSGU<FG_ELEMENT> > subspaces_;
+  std::vector<SubspaceSGU<FG_ELEMENT> > subspaces_; // subspaces of the dsg
+
+  std::vector<FG_ELEMENT> subspacesData_; // allows linear access to all subspaces data
```

Comment:
@mhurler123 I recall there was a reason why this additional vector was introduced, but I can't recall it. Do you remember?

Reply:
I remember that Mario copied the subspaces to an extra buffer to allow reducing them with a single MPI call. This effectively doubled the memory consumption per worker during the combination step. Note that the SubspaceSGU in my implementation only hold pointers to the subspacesData_.

See https://github.com/SGpp/DisCoTec/blob/master/distributedcombigrid/src/sgpp/distributedcombigrid/combicom/CombiCom.hpp#L776 on master:

Reply:
Great, thank you! I think this is the way to go for now.